source/common/singleton/manager_impl.cc

```diff
@@ -0,0 +1,26 @@
+#include "common/singleton/manager_impl.h"
+
+#include "envoy/registry/registry.h"
+
+namespace Envoy {
+namespace Singleton {
+
+InstancePtr ManagerImpl::get(const std::string& name) {
+  verifyRegistration(name);
+  return singletons_[name].lock();
+}
+
+void ManagerImpl::set(const std::string& name, InstancePtr singleton) {
+  verifyRegistration(name);
+  singletons_[name] = singleton;
+}
+
+void ManagerImpl::verifyRegistration(const std::string& name) {
+  if (nullptr == Registry::FactoryRegistry<Registration>::getFactory(name)) {
```

Comment:
Should this be an `ASSERT`? I.e., is this something that could happen if not programmer error.